packages/terra-navigation-layout/examples/IndexExampleTemplate.jsx

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable import/no-extraneous-dependencies, import/no-webpack-loader-syntax, import/first, import/no-unresolved, import/extensions */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Markdown from 'terra-markdown';
+import SlidePanel from 'terra-slide-panel';
+
+const propTypes = {
+  example: PropTypes.element,
+  exampleSrc: PropTypes.string,
+  title: PropTypes.string,
+  children: PropTypes.element,
+};
+
+const defaultProps = {
+  example: null,
+  exampleSrc: null,
+  title: null,
+};
+
+const codeIndicator = '```';
+
+class IndexExampleTemplate extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { panelIsOpen: false };
+
+    this.handlePanelToggle = this.handlePanelToggle.bind(this);
+  }
+
+  handlePanelToggle() {
+    this.setState({ panelIsOpen: !this.state.panelIsOpen });
+  }
+
+  render() {
+    const { title, example, exampleSrc, children } = this.props;
+
+    const indexExample = (
+      <div>
+        <button onClick={this.handlePanelToggle} style={{ display: 'inline-block' }}>Show Source Code</button>
```

Comment:
Do we have a documented convention on when to use inline styles vs classes? We get a lot of questions from consumers on this, and it'd be great if we have a source of truth to pass to consumers. Otherwise, you'll have different developers both on and off Terra that will give an opinion that may not match the coded examples.